### ISSUE TITLE
Using Pre-trained Checkpoints for Mouse Dataset

### ISSUE THREAD

**Issue by Addison-W-ad** [GET-Foundation/get_model#11]
Hi Xi,

Thank you for sharing this great work! I am working with a mouse dataset and would like to know if I could use one of the pre-trained checkpoints (checkpoint-799.pth or checkpoint-best.pth) as input for my analysis. I also noticed that similar questions were raised in closed issues that you mentioned pre-trained mouse checkpoints based on the mouse sci-ATAC. Where I can find these pre-trained checkpoints or do you recommend a different approach?

Thank you!

**Comment by fuxialexander** [comment]
For now I would suggest directly finetune from one of the human checkpoint. The mouse checkpoint was not trained and test extensively, only trained on ATAC and also kind of tied to my collaborator's project, so it's kinda premature to release it. Meanwhile we are try to working a jointly trained model. Please stay tuned.
Note that you'd need to use the [mouse motif](https://resources.altius.org/~jvierstra/projects/motif-clustering/releases/v1.0) to process your data. Most of the motif cluster is shared between human and mouse so finetuning should be fine.

**Comment by Addison-W-ad** [comment]
Thanks for this quick response! It was very helpful. 

I’m also encountering an issue when running the predict_acat.ipynb training without pretraining using 

```
cfg = load_config('finetune_tutorial.yaml')
cfg.model = load_config('model/GETRegionFinetuneATAC').model.model
cfg.finetune.checkpoint = None
trainer = run(cfg). 
```

The error I receive is:
RuntimeError:` mat1 and mat2 shapes cannot be multiplied (1600x234 and 283x768) 

My data was processed for mouse motif following the prepare_pbmc.ipynb workflow.  I’m wondering if I can still use the finetune_tutorial.yaml file directly or how I might leverage it to accommodate the mouse data.

Thank you in advance for any suggestions!

**Comment by fuxialexander** [comment]
Yes I think the mouse motif file has some of the cluster removed. Two way to workaround this: when generating the motif zarr, use the human motif id list to initialize the columns and set the missing motif values to 0. Or, use the mouse motifs only, but set the num_motif parameter in config (see https://github.com/GET-Foundation/get_model/blob/master/get_model/config/model/GETRegionFinetuneATAC.yaml) to 234.

**Comment by fuxialexander** [comment]
My modified `create_peak_motif` for the first workaround (note that I added a peak-length normalization here which is not in the original code, as the original peak used in the paper is all 400 bp). Not really sure whether normalization or not is better yet though.

 
```
import zarr
import pandas as pd
import numpy as np
def create_peak_motif(peak_motif_bed, output_zarr, peak_bed):
    """
    Create a peak motif zarr file from a peak motif bed file.

    This function reads a peak motif bed file, pivots the data, and saves it to a zarr file.
    The zarr file contains three datasets: 'data', 'peak_names', 'motif_names', and 'accessibility'.
    The 'data' dataset is a sparse matrix containing the peak motif data.
    The 'peak_names' dataset contains the peak names.
    The 'motif_names' dataset contains the motif names.

    Args:
        peak_motif_bed (str): Path to the peak motif bed file.
        output_zarr (str): Path to the output zarr file.
    """
    import pandas as pd
    motif_annotations = pd.read_excel('https://resources.altius.org/~jvierstra/projects/motif-clustering/releases/v1.0/motif_annotations.xlsx')
    motif_cluster_ids = motif_annotations.Name.unique()
    # Read the peak motif bed file
    peak_motif = pd.read_csv(
        peak_motif_bed,
        sep="\t",
        header=None,
        names=["Chromosome", "Start", "End", "Motif_cluster", "Score"],
    )

    # Pivot the data
    peak_motif_pivoted = peak_motif.pivot_table(
        index=["Chromosome", "Start", "End"],
        columns="Motif_cluster",
        values="Score",
        fill_value=0,
    )

    peak_motif_pivoted.reset_index(inplace=True)
    # add missing motif columns
    for motif_cluster_id in motif_cluster_ids:
        if motif_cluster_id not in peak_motif_pivoted.columns:
            peak_motif_pivoted[motif_cluster_id] = 1
    # Create the 'Name' column
    peak_motif_pivoted["Name"] = peak_motif_pivoted.apply(
        lambda x: f'{x["Chromosome"]}:{x["Start"]}-{x["End"]}', axis=1
    )
    peak_motif_pivoted = peak_motif_pivoted.drop(columns=["Chromosome", "Start", "End"])

    # Read the original peak bed file
    original_peaks = pd.read_csv(
        peak_bed, sep="\t", header=None, names=["Chromosome", "Start", "End", "Score"]
    )

    # exclude chrM and chrY
    original_peaks = original_peaks[~original_peaks.Chromosome.isin(["chrM", "chrY"])]
    original_peaks["Name"] = original_peaks.apply(
        lambda x: f'{x["Chromosome"]}:{x["Start"]}-{x["End"]}', axis=1
    )
    
    new_columns = list(motif_cluster_ids) + ["Name"]

    # sort motif columns
    peak_motif_pivoted = peak_motif_pivoted[new_columns]

    # Merge the pivoted data with the original peaks
    merged_data = pd.merge(original_peaks, peak_motif_pivoted, on="Name", how="left")

    # Fill NaN values with 0 for motif columns
    motif_columns = [
        col
        for col in merged_data.columns
        if col not in ["Chromosome", "Start", "End", "Score", "Name"]
    ]
    
    merged_data[motif_columns] = merged_data[motif_columns].fillna(0)
    peak_length = (merged_data.End - merged_data.Start).values / 400 # convert to kb
    merged_data[motif_columns] = merged_data[motif_columns].div(peak_length, axis=0)
    # Prepare data for zarr storage
    name_values = list(merged_data["Name"].values)
    motif_values = motif_columns

    # Create sparse matrix
    motif_data_matrix = merged_data[motif_columns].values
    # Open zarr store and save data
    from numcodecs import Blosc

    z = zarr.open(output_zarr, mode="w")
    z.create_dataset(
        "data",
        data=motif_data_matrix.data,
        chunks=(1000, motif_data_matrix.shape[1]),
        dtype=np.float32,
        compressor=Blosc(cname="zstd", clevel=3, shuffle=Blosc.BITSHUFFLE),
        shape=motif_data_matrix.shape,
    )
    z.create_dataset("peak_names", data=name_values)
    z.create_dataset("motif_names", data=motif_values)

    print(f"Peak motif data saved to {output_zarr}")

create_peak_motif(get_motif_output, "mouse_mm10.zarr", peak_bed) # all cell types will later be added to the same zarr file as we use the same peak set.
```

Lines added for normalization is
```
peak_length = (merged_data.End - merged_data.Start).values / 400 # convert to kb
merged_data[motif_columns] = merged_data[motif_columns].div(peak_length, axis=0)
```

**Comment by biubiu-1** [comment]
Thanks for the helpful discussion in this thread!

Just a quick follow-up question — if we replace the original human motif set with a mouse-specific motif collection (e.g., 234 motifs instead of 283), does that mean we also need to re-run the pretraining stage, since the motif embedding layer would be re-initialized? Or is it still possible to fine-tune directly using the pretrained human model in this case?

Appreciate any guidance!

**Comment by fuxialexander** [comment]
I think you can still finetune, although it might be less optimal.
